src/widgets/survey/model/services/tests/CollectCompletionsService.collectAll.test.ts

```diff
@@ -0,0 +1,294 @@
+    it('Should return empty array when "exclude" is set to the other path and flow record is null', () => {
+      const getFlowMock = jest
+        .spyOn(storageHelpers, 'getFlowRecord')
+        .mockReturnValue(null);
+
+      const progress = getRegularProgressRecord(pathOne);
+
+      progress.payload.availableTo = new Date(2023, 3, 7).getTime();
+
+      const service = new CollectCompletionsService([progress]);
+
+      const result = service.collectAll({
+        ...pathOne,
+        ...excluded,
+      });
+
+      expect(getFlowMock).toBeCalledTimes(1);
+
+      expect(result).toEqual([]);
+    });
+  });
+
+  it('Should return empty array when "exclude" is not set and availableTo is not set', () => {
+    const progress = getRegularProgressRecord(pathOne);
+
+    progress.payload.availableTo = null;
+
+    const service = new CollectCompletionsService([progress]);
+
+    const result = service.collectAll(pathOne);
+
+    expect(result).toEqual([]);
+  });
+
+  it('Should return empty array when availableTo is set and flow record is null', () => {
+    const getFlowMock = jest
+      .spyOn(storageHelpers, 'getFlowRecord')
+      .mockReturnValue(null);
+
+    const progress = getRegularProgressRecord(pathOne);
+
+    progress.payload.availableTo = new Date(2023, 3, 7).getTime();
+
+    const service = new CollectCompletionsService([progress]);
+
+    const result = service.collectAll();
+
+    expect(getFlowMock).toBeCalledTimes(1);
+
+    expect(result).toEqual([]);
+  });
+
+  it('Should return empty array when isEntityExpired is false', () => {
+    jest.spyOn(storageHelpers, 'getFlowRecord').mockReturnValue({} as any);
+
+    const isEntityExpiredMock = jest
+      .spyOn(survey, 'isEntityExpired')
+      .mockReturnValue(false);
+
+    const firstProgress = getRegularProgressRecord(pathOne);
+    const secondProgress = getRegularProgressRecord(pathTwo);
+
+    firstProgress.payload.availableTo = new Date(2023, 3, 7).getTime();
+    secondProgress.payload.availableTo = new Date(2023, 3, 7).getTime();
+
+    const service = new CollectCompletionsService([
+      firstProgress,
+      secondProgress,
+    ]);
+
+    const result = service.collectAll();
+
+    expect(isEntityExpiredMock).toBeCalledTimes(2);
+
+    expect(result).toEqual([]);
+  });
+
+  it('Should return two collect items when isEntityExpired is true', () => {
+    jest.spyOn(storageHelpers, 'getFlowRecord').mockReturnValue({} as any);
+
+    jest.spyOn(survey, 'isEntityExpired').mockReturnValue(true);
+
+    const firstProgress = getRegularProgressRecord(pathOne);
+    const secondProgress = getRegularProgressRecord(pathTwo);
+
+    firstProgress.payload.availableTo = new Date(2023, 3, 7).getTime();
+    secondProgress.payload.availableTo = new Date(2023, 3, 7).getTime();
+
+    const service = new CollectCompletionsService([
+      firstProgress,
+      secondProgress,
+    ]);
+
+    //@ts-expect-error
+    service.collect = jest
```

Comment:
Is there a reason to mock the method? I can see it as an important part of the `CollectCompletionsService`, not just a utility method like, for instance, `getNow`

Reply:
Yes, it is. The "collect" method and the methods-chain initiated by collect - has been already covered by the tests of another public function: collectForEntity. So, no need to do double work and to make tests more complex. Coverage report for this module shows 100%.